catalog/core/catalog-core-dynamic/src/main/java/ddf/catalog/data/dynamic/api/MetacardFactory.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.data.dynamic.api;
+
+import org.apache.commons.beanutils.LazyDynaClass;
+
+import ddf.catalog.data.dynamic.impl.DynamicMetacard;
+import ddf.catalog.data.dynamic.impl.MetacardPropertyDescriptor;
+
+public interface MetacardFactory {
+    /**
+     * Creates an instance of a basic dynamic metacard with the default attributes.
+     * @return new instance of a basic dynamic metacard
+     */
+    DynamicMetacard newInstance();
+
+    /**
+     * Creates an instance of a dynamic metacard corresponding to the given name. It is
+     * expected that a metacard of the given name has been registered with the factory
+     * beforehand.
+     * @param name the name of the metacard type to create
+     * @return an instance of a dynamic metacard of the type whose name was provided
+     */
+    DynamicMetacard newInstance(String name);
```

Comment:
"It is expected that a metacard of the given name has been registered with the factory beforehand."
This would be done via the "addDynaClass()" method, right?  Maybe states that explicitly?  Also, what happens if the class isn't registered?  Should this method throw an IllegalArguementException like the beanutils classes do?